Pr92x L1T calo, muon, uGT for  2017 (v93.6 minus v93.3,v92.24)

rekovic opened the pull request:
92x PR meant to go into 9_1_0.
Total of 86 files, but big part of it are config files (37).

It includes updates for 2017 in Calo, Muon and uGT.

- Calo calibrations 2017 caloStage2Params v1_v4 &  v1_5. Letter set as default.
  - Layer2 new EG/Tau Iso, Jet, MET.  New feature MET PUS.
  - Layer1 support for HEP17 
  - Needed LUT externals are in cms-data/L1Trigger-L1TCalorimeter#17
- Muon updates for 2017
  - uGMT update of tracking of FW version (0x4010000) and (0x5000000).
  - uGMT intermediate unpacker (relevant in DQM, not full emulator)
  - uGMT unpacker fix problem of deleting all blocks in case of ZeroSupression
- uGT
  - Indexing and overlap-removal 

It is equivalent to l1t-inegration-v93.6 minus two tags: v93.3 (omtf unpacker) and v92.24 (L1T Legacy Ntuple)  which are both reverted.  
 - Tag 93.3 needs to be blessed with efficiency plots. 
 - Tag 92.24 need a fix to run on run1 and run2_2105. 
So for now, they are both out.

bortigno committed:
Added customisation to include legacy and upgrade in the same sequence

bortigno committed:
Adding token for the legacy muons

bortigno committed:
Adding legacy muon to the ntuples

bortigno committed:
running configuration

bortigno committed:
first running version with legacy muons appearing in the ntuples

bortigno committed:
fixing conversion. there was a missing charge_valid when creating the muon candidate from the legacy information.

bortigno committed:
changes in some of the digi source

bortigno committed:
add the legacy settings to RAW ntupliser

bortigno committed:
producing rates across all eta but probably not kosher

bortigno committed:
use physical values instead of hardware values for pt,eta, and phi. clean up of comments.

bortigno committed:
removing track finder index for the legacy muons as it uses a different convention

bortigno committed:
clean up from comments - still crash with inputTag in DTTF

bortigno committed:
adding twinmux unpacker in RAW2DIGI

bortigno committed:
adding an input tag to the dttf track finder emulator to be able to re-emualte from twinmux digis. not compatible with run1 and run2_2105.

bortigno committed:
legacy muon re-emulation update: module working with the modified version of the DTTF emulator. not re-emulating trigger primitives.

dntaylor committed:
Begin to add phi dependent layer1 LUTs

dntaylor committed:
add phi dependent support for layer1 calibrations

Conflicts:
	L1Trigger/Configuration/python/customiseSettings.py

dntaylor committed:
updated calibration access

dntaylor committed:
fix again

dntaylor committed:
add backwards support for previous calo params version

dntaylor committed:
add new et binning to layer1 phi dependent

dntaylor committed:
removing old

thomreis committed:
Set correct uGMT and uGT FW version for RAW format change

thomreis committed:
Set correct uGMT FW version for extrapolation LUT format changes

thomreis committed:
New cancel out windows. 'Aggressive' tuning of 3 April 2017.

thomreis committed:
Fix FatalRootError exception with GEN tree.

thomreis committed:
Fix uGMT intermediate muon unpacker bug that fills muons in the wrong collection.

bundocka committed:
Implement MET pileup subtraction with tower Et threshold LUT, add LUT generator & LUT

bundocka committed:
Changes to caloParams helper

rekovic committed:
Document FW version numbers in GMT and GT unpackers.

rekovic committed:
Introduce Indexing: 1) expand Calo and Muon Templates, 2) expand class ConditionEvaluation for method checkIndex(), 3) expand method checkObjectParameter in Calo and MuCondtions to call checkIndexing and cut on it.

rekovic committed:
New class CorrelationWithOverlapRemovalTemplate, to hold information for new CorrelationWithOverlapRemoval type algorithms parsed from xml file.

rekovic committed:
Extend TriggerMenuParser to parse Indexing and CorrelationWithOverlapRemoval.

rekovic committed:
Define new GT condition and type (CondCorrelationWithOverlapRemoval,Type2corWithOverlapRemoval).

rekovic committed:
New class to implement CorrWithOverlapRemovalCondition.

rekovic committed:
Update TriggerMenu and GT Logic board to produce map of objects and conditions for new corrWithOverlapRemoval algorithms.

rekovic committed:
Spell filename correctly.

rekovic committed:
Example menu xml file for testing indexing and overlap removal in correlation conditions.

rekovic committed:
In headers document changes introduced.

thomreis committed:
Set default BX range in case the zero suppression deletes all the data blocks and the BX cannot be reconstructed from them.

thomreis committed:
simplifying string concatenation

thomreis committed:
Add includes.

thomreis committed:
Fix potential exception if BX != 0.

abeschi committed:
Corrected shape Id LUT

bortigno committed:
updating reEmu with 900 integration branch version

konec committed:
OMTF: first version of unpacker.
Works for all OMTF RAW inputs and outputs (i.e. rpc raws, dt stubs, csc lcts, omtf muons)
Currently, by default reads some cabling information from txt file, which is placed in data directory.
However, possibility to read it from DB is also implemented and can be switched in config.

rekovic committed:
Fix compilation errors introduced in PR 498 (tag v92.14): changed methods in CaloParamsHelper needed to be propagated to L1TCaloParamsOnlineProd.

rekovic committed:
Post rebase fix of customiseSettings of caloParams.

rekovic committed:
Post rebase fix: add missed file caloStage2Params_2017_v1_4_cfi.py.

rekovic committed:
Rebase fix to Stage2Layer2EtSumAlgorithmFirmwareImp1 and Stage2Layer2JetAlgorithmFirmwareImp1.

rekovic committed:
Add L1TCalorimeter data.

bundocka committed:
Met pileup subtraction LUT update

bundocka committed:
Remove cout statement

rekovic committed:
Set caloParams_2017_v1_4 as default in hackConditions.

thomreis committed:
Unpack OMTF track address bits.

thomreis committed:
Merged l1t-integration-v93.2 from repository cms-l1t-offline

bundocka committed:
Merged l1t-integration-v93.2 from repository cms-l1t-offline

bundocka committed:
Update MET pileup sub LUT

bundocka committed:
Add new caloParams for new MET LUT v3

bundocka committed:
Update to MET LUT

bundocka committed:
EtSum emu bugfix when no pileup subtraction type given from caloParams

bundocka committed:
EtSum emu fix

rekovic committed:
OMTF unpacker: rename module name from omtfDigis to omtfStage2Digis to follow convention. Include omtfStage2Digis in the RAW2DIGI sequence.

rekovic committed:
Merge branch 'uGMT-unpacker_omtf-track-address-unpacking_91x' of https://github.com/thomreis/cmssw into thomreis-uGMT-unpacker_omtf-track-address-unpacking_91x

rekovic committed:
Merge branch 'thomreis-uGMT-unpacker_omtf-track-address-unpacking_91x' into l1t-integration-CMSSW_9_1_0_pre3

rekovic committed:
Merge branch 'uGMT-calosumproducer-BX-range-fix_91x' of https://github.com/thomreis/cmssw into thomreis-uGMT-calosumproducer-BX-range-fix_91x

rekovic committed:
Merge branch 'thomreis-uGMT-calosumproducer-BX-range-fix_91x' into l1t-integration-CMSSW_9_1_0_pre3

rekovic committed:
Merge branch 'metLUTUpdate910pre3' of https://github.com/bundocka/cmssw into bundocka-metLUTUpdate910pre3

rekovic committed:
Merge branch 'bundocka-metLUTUpdate910pre3' into l1t-integration-CMSSW_9_1_0_pre3

bundocka committed:
Re-add v2 MET LUT

bundocka committed:
Re-add v1 MET LUT

rekovic committed:
Merge branch 'metLUTUpdate910pre3' of https://github.com/bundocka/cmssw into bundocka-metLUTUpdate910pre3

rekovic committed:
Merge branch 'bundocka-metLUTUpdate910pre3' into l1t-integration-CMSSW_9_1_0_pre3

rekovic committed:
Make default caloParams 2017_v1_5.

rekovic committed:
Revert "OMTF unpacker: rename module name from omtfDigis to omtfStage2Digis to follow convention. Include omtfStage2Digis in the RAW2DIGI sequence."

This reverts commit e466296b0619658f5dfe0e54d6b125fb67bbd1fa.

rekovic committed:
Revert "OMTF: first version of unpacker."

This reverts commit 69b9261dfcead68892ad48d65ee4227f78f9ddcd.

rekovic committed:
Remove Calo data files and one Global data files used for testing OverlapRemoval.

cmsbuild commented:
A new Pull Request was created by  @rekovic  for master.

It involves the following packages:

CondFormats/L1TObjects
DataFormats/L1TMuon
EventFilter/L1TRawToDigi
L1Trigger/Configuration
L1Trigger/DTTrackFinder
L1Trigger/L1TCaloLayer1
L1Trigger/L1TCalorimeter
L1Trigger/L1TCommon
L1Trigger/L1TGlobal
L1Trigger/L1TMuon
L1Trigger/L1TNtuples
L1TriggerConfig/L1TConfigProducers


@ghellwig, @arunhep, @cerminar, @cmsbuild, @rekovic, @franzoni, @ggovi, @mmusich, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @ghellwig, @kreczko, @tocheng this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

Martin-Grunewald commented:
cms-sw:master now refers to 92X. If you want this in 91X, need to make a PR against 91X as well.

Also, does this superseed v93.2 you describe on the L1T wiki https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideL1TStage2Instructions ?

rekovic commented:
@Martin-Grunewald  Thanks.

Relevant to the emulator and compared to v93.2 it has an update to MET PUS LUTs.
The twiki is now updated.

rekovic commented:
please test with cms-sw/cmsdist#3015

cmsbuild commented:
Ignoring test request.
**ERROR**: Unable to find cmsdist Pull request cms-sw/cmsdist#3015

rekovic commented:
please test with cms-sw/cmsdist#3017

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3017

cmsbuild commented:
-1

Tested at: bd27ad92b6f0cfb6fa961561ce57b814ff959ed5


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/aa055a6ee183e4d9e8d1f98ab6cccbc1b94e14ae
https://github.com/cms-sw/cmssw/commit/2a1a421070a0073af91ff7aa7f4d00e2240da12d
https://github.com/cms-sw/cmssw/commit/778592f7475a9a59f9f11222694bbc1b0b4187d6
https://github.com/cms-sw/cmssw/commit/35eabe76fbd10cd4bc10869adcb02a93256fdfd3
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19648/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19648/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19648/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

</pre>4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>140.53 step2
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

</pre>9.0 step2
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step2_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>25.0 step2
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>50202.0 step2
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step2_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py TTbar_8TeV_TuneCUETP8M1_cfi  --conditions auto:run1_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot Realistic8TeVCollision : FAILED - time: date Mon May  8 20:24:27 2017-date Mon May  8 20:20:43 2017 s - exit: 17920
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_25ns --magField 38T_PostLS1 : FAILED - time: date Mon May  8 20:24:28 2017-date Mon May  8 20:20:56 2017 s - exit: 17920
cmsDriver.py TTbar_Tauola_8TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW  --mc --scenario=pp -n 10 --conditions auto:run1_mc_Fake --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --fileout file:RelVal_Raw_Fake_MC.root : FAILED - time: date Mon May  8 20:34:26 2017-date Mon May  8 20:26:03 2017 s - exit: 17920
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake1 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_25ns --fileout file:RelVal_Raw_Fake1_MC.root : FAILED - time: date Mon May  8 20:36:25 2017-date Mon May  8 20:29:30 2017 s - exit: 17920



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/aa055a6ee183e4d9e8d1f98ab6cccbc1b94e14ae
https://github.com/cms-sw/cmssw/commit/2a1a421070a0073af91ff7aa7f4d00e2240da12d
https://github.com/cms-sw/cmssw/commit/778592f7475a9a59f9f11222694bbc1b0b4187d6
https://github.com/cms-sw/cmssw/commit/35eabe76fbd10cd4bc10869adcb02a93256fdfd3
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19648/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19648/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

smuzaffar commented:
please test with cms-sw/cmsdist#3017

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3017

cmsbuild commented:
-1

Tested at: bd27ad92b6f0cfb6fa961561ce57b814ff959ed5


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/aa055a6ee183e4d9e8d1f98ab6cccbc1b94e14ae
https://github.com/cms-sw/cmssw/commit/2a1a421070a0073af91ff7aa7f4d00e2240da12d
https://github.com/cms-sw/cmssw/commit/778592f7475a9a59f9f11222694bbc1b0b4187d6
https://github.com/cms-sw/cmssw/commit/35eabe76fbd10cd4bc10869adcb02a93256fdfd3
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19654/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19654/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19654/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

</pre>4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>140.53 step2
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

</pre>9.0 step2
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step2_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>25.0 step2
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>50202.0 step2
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step2_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py TTbar_8TeV_TuneCUETP8M1_cfi  --conditions auto:run1_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot Realistic8TeVCollision : FAILED - time: date Tue May  9 02:24:47 2017-date Tue May  9 02:21:13 2017 s - exit: 17920
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_25ns --magField 38T_PostLS1 : FAILED - time: date Tue May  9 02:24:47 2017-date Tue May  9 02:21:28 2017 s - exit: 17920
cmsDriver.py TTbar_Tauola_8TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW  --mc --scenario=pp -n 10 --conditions auto:run1_mc_Fake --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --fileout file:RelVal_Raw_Fake_MC.root : FAILED - time: date Tue May  9 02:33:45 2017-date Tue May  9 02:25:46 2017 s - exit: 17920
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake1 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_25ns --fileout file:RelVal_Raw_Fake1_MC.root : FAILED - time: date Tue May  9 02:35:58 2017-date Tue May  9 02:29:53 2017 s - exit: 17920



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/aa055a6ee183e4d9e8d1f98ab6cccbc1b94e14ae
https://github.com/cms-sw/cmssw/commit/2a1a421070a0073af91ff7aa7f4d00e2240da12d
https://github.com/cms-sw/cmssw/commit/778592f7475a9a59f9f11222694bbc1b0b4187d6
https://github.com/cms-sw/cmssw/commit/35eabe76fbd10cd4bc10869adcb02a93256fdfd3
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19654/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19654/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

rekovic committed:
Revert "legacy muon re-emulation update: module working with the modified version of the DTTF emulator. not re-emulating trigger primitives."

This reverts commit bd030e26218f819e3a274f2a51a75d5f915ce157.

rekovic committed:
Revert "adding an input tag to the dttf track finder emulator to be able to re-emualte from twinmux digis. not compatible with run1 and run2_2105."

This reverts commit cce4ba290d08db032b78083182222ce87decb3d0.

cmsbuild commented:
Pull request #18622 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @rekovic, @franzoni, @ggovi, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

rekovic commented:
reverted tag v92.24 (Ntupling of Legacy muons and changes introduced to DTTrackFinder --> not compatible with run1 and run2_2105)

rekovic commented:
please test with cms-sw/cmsdist#3017

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3017

cmsbuild commented:
+1
Tested at: f801fe09f99d24294de3b53e546eacb50cf19b6a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19668/summary.html

cmsbuild commented:
Comparison job queued.

rekovic commented:
+1

davidlange6 commented on L1Trigger/L1TGlobal/plugins/TriggerMenuParser.cc:
```diff
@@ -2853,6 +2959,338 @@ bool l1t::TriggerMenuParser::parseCorrelation(
+      }
+      
+// loop over legs      
+      for (size_t jj = 0; jj < objects.size(); jj++)
```
hi @rekovic - use a for (auto const & object : objects) here

rekovic replied:
In this case the auto for loop is not appropriate as the index jj is used to set values in other auxiliary vectors.

davidlange6 commented on L1Trigger/L1TGlobal/src/CorrWithOverlapRemovalCondition.cc:
```diff
@@ -0,0 +1,2135 @@
+                    phi1Phy = 0.5*(binEdges.second + binEdges.first);			
+                    binEdges = m_gtScales->getJETScales().etaBins.at(etaBin1);
+                    eta1Phy = 0.5*(binEdges.second + binEdges.first);		
+                    //CRASHES HERE:
```
perhaps this comment is now obsolete?

rekovic replied:
yes. removed.

davidlange6 commented on L1Trigger/L1TCaloLayer1/plugins/L1TCaloLayer1.cc:
```diff
@@ -81,9 +81,13 @@ class L1TCaloLayer1 : public edm::EDProducer {
-  std::vector< std::vector< std::vector < uint32_t > > > ecalLUT;
-  std::vector< std::vector< std::vector < uint32_t > > > hcalLUT;
-  std::vector< std::vector< uint32_t > > hfLUT;
+  std::vector< std::vector< std::vector< std::vector < uint32_t > > > > ecalLUT;
```
Hi @rekovic - i had problems with 3 layers of vectors, now you have four:(

so what are these things? it seems some have well understood (eg, hardwired) sizes.

rekovic replied:
There are CaloLayer1 Tower calibrations in bins of pt, eta, and now also of phi.
Will add comment.

davidlange6 commented on L1Trigger/L1TCaloLayer1/src/L1TCaloLayer1FetchLUTs.cc:
```diff
@@ -115,115 +129,159 @@ bool L1TCaloLayer1FetchLUTs(const edm::EventSetup& iSetup,
+        else {
+          iPhi = (index-18)*4+1;
+        }
+        if (abs(caloEta) == 41 && iPhi<3) iPhi = 3;
```
here and elsewhere, use std::abs

rekovic replied:
done

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19668/summary.html

Comparison Summary:
 - You potentially added 5833 lines to the logs
 - Reco comparison results: 1689 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1831050
 - DQMHistoTests: Total failures: 57856
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1773014
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

rekovic committed:
Add comment to describe 4 layers of vectors.

rekovic committed:
Remove obsolete comment.

rekovic committed:
Use std::abs instead of abs.

cmsbuild commented:
Pull request #18622 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @rekovic, @franzoni, @ggovi, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

rekovic commented:
@davidlange6 
Follow up on your question https://github.com/cms-sw/cmssw/pull/18622#issuecomment-300463411.

Example of ecal and hcal the Layer1 Calib values are provided in  eta (28 towers)  x  2 sides x phi (256).
Will document this also.

rekovic commented:
no pT previously stated.

rekovic committed:
Correct comment to describe four layers of vectors in Layer1.

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #18622 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @rekovic, @franzoni, @ggovi, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

cmsbuild commented:
-1

Tested at: 1418195478ee6e9ece0864d9b835c5e31c317c55

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19722/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
136.731 step2
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step2_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>10042.0 step2
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step2_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>10024.0 step2
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>10824.0 step2
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018.log

</pre>21234.0 step2
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D4_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D4+RecoFullGlobal_2023D4+HARVESTFullGlobal_2023D4/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D4_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D4+RecoFullGlobal_2023D4+HARVESTFullGlobal_2023D4.log

</pre>23234.0 step2
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D8_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D8+RecoFullGlobal_2023D8+HARVESTFullGlobal_2023D8/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D8_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D8+RecoFullGlobal_2023D8+HARVESTFullGlobal_2023D8.log

</pre>24434.0 step2
<pre>runTheMatrix-results/24434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D12_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D12+RecoFullGlobal_2023D12+HARVESTFullGlobal_2023D12/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D12_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D12+RecoFullGlobal_2023D12+HARVESTFullGlobal_2023D12.log

</pre>26634.0 step2
<pre>runTheMatrix-results/26634.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D15_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D15+RecoFullGlobal_2023D15+HARVESTFullGlobal_2023D15/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D15_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D15+RecoFullGlobal_2023D15+HARVESTFullGlobal_2023D15.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_Fake2 --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --fileout file:RelVal_Raw_Fake2_DATA.root --filein /store/data/Run2015D/MuonEG/RAW/v1/000/256/677/00000/80950A90-745D-E511-92FD-02163E011C5D.root : FAILED - time: date Wed May 10 17:48:55 2017-date Wed May 10 17:48:15 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake2.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:48:55 2017-date Wed May 10 17:48:15 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_DATA.root --fileout file:RelVal_Raw_Fake2_DATA_HLT_RECO.root : FAILED - time: date Wed May 10 17:48:55 2017-date Wed May 10 17:48:15 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_PRef --relval 9000,50 --datatier "RAW" --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --eventcontent RAW --fileout file:RelVal_Raw_PRef_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Wed May 10 17:49:11 2017-date Wed May 10 17:48:17 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:49:11 2017-date Wed May 10 17:48:17 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_DATA.root --fileout file:RelVal_Raw_PRef_DATA_HLT_RECO.root : FAILED - time: date Wed May 10 17:49:11 2017-date Wed May 10 17:48:17 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full2015Data --data --scenario=HeavyIons -n 10 --conditions auto:run2_hlt_HIon --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016,Run2_HI --magField 38T_PostLS1 --fileout file:RelVal_Raw_HIon_DATA.root --filein /store/hidata/HIRun2015/HIHardProbes/RAW-RECO/HighPtJet-PromptReco-v1/000/263/689/00000/1802CD9A-DDB8-E511-9CF9-02163E0138CA.root : FAILED - time: date Wed May 10 17:49:04 2017-date Wed May 10 17:48:18 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:49:04 2017-date Wed May 10 17:48:18 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --data --scenario=HeavyIons -n 10 --conditions auto:run2_data_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016,Run2_HI --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_HIon_DATA.root --fileout file:RelVal_Raw_HIon_DATA_HLT_RECO.root : FAILED - time: date Wed May 10 17:49:04 2017-date Wed May 10 17:48:18 2017 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun_MC.root : FAILED - time: date Wed May 10 17:49:22 2017-date Wed May 10 17:48:23 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:49:22 2017-date Wed May 10 17:48:23 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Wed May 10 17:49:22 2017-date Wed May 10 17:48:23 2017 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_PRef_MC.root : FAILED - time: date Wed May 10 17:49:19 2017-date Wed May 10 17:48:28 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:49:19 2017-date Wed May 10 17:48:28 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root : FAILED - time: date Wed May 10 17:49:19 2017-date Wed May 10 17:48:28 2017 s - exit: 21504
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_2016 --magField 38T_PostLS1 : FAILED - time: date Wed May 10 17:49:49 2017-date Wed May 10 17:49:01 2017 s - exit: 23040
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun2016 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun2016_MC.root : FAILED - time: date Wed May 10 17:51:57 2017-date Wed May 10 17:49:04 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun2016.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:51:57 2017-date Wed May 10 17:49:04 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun2016,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun2016 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun2016_MC.root --fileout file:RelVal_Raw_GRun2016_MC_HLT_RECO.root : FAILED - time: date Wed May 10 17:51:57 2017-date Wed May 10 17:49:04 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_PIon --relval 9000,50 --datatier "RAW" --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --eventcontent RAW --fileout file:RelVal_Raw_PIon_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Wed May 10 17:49:57 2017-date Wed May 10 17:49:06 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:49:57 2017-date Wed May 10 17:49:06 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_DATA.root --fileout file:RelVal_Raw_PIon_DATA_HLT_RECO.root : FAILED - time: date Wed May 10 17:49:57 2017-date Wed May 10 17:49:06 2017 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016,Run2_HI --magField 38T_PostLS1 --fileout file:RelVal_Raw_HIon_MC.root : FAILED - time: date Wed May 10 17:50:06 2017-date Wed May 10 17:49:23 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:50:06 2017-date Wed May 10 17:49:23 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016,Run2_HI --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_HIon_MC.root --fileout file:RelVal_Raw_HIon_MC_HLT_RECO.root : FAILED - time: date Wed May 10 17:50:06 2017-date Wed May 10 17:49:23 2017 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake2 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --fileout file:RelVal_Raw_Fake2_MC.root : FAILED - time: date Wed May 10 17:50:15 2017-date Wed May 10 17:49:24 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake2.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:50:15 2017-date Wed May 10 17:49:24 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_MC.root --fileout file:RelVal_Raw_Fake2_MC_HLT_RECO.root : FAILED - time: date Wed May 10 17:50:15 2017-date Wed May 10 17:49:24 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_GRun --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Wed May 10 17:52:54 2017-date Wed May 10 17:50:22 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:52:54 2017-date Wed May 10 17:50:22 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_DATA.root --fileout file:RelVal_Raw_GRun_DATA_HLT_RECO.root : FAILED - time: date Wed May 10 17:52:54 2017-date Wed May 10 17:50:22 2017 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_PIon_MC.root : FAILED - time: date Wed May 10 17:53:12 2017-date Wed May 10 17:51:46 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:53:12 2017-date Wed May 10 17:51:46 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_MC.root --fileout file:RelVal_Raw_PIon_MC_HLT_RECO.root : FAILED - time: date Wed May 10 17:53:12 2017-date Wed May 10 17:51:46 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_GRun2016 --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun2016_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Wed May 10 17:54:15 2017-date Wed May 10 17:51:58 2017 s - exit: 23040
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_9_2_X_2017-05-09-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun2016.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 10 17:54:15 2017-date Wed May 10 17:51:58 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun2016,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun2016 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun2016_DATA.root --fileout file:RelVal_Raw_GRun2016_DATA_HLT_RECO.root : FAILED - time: date Wed May 10 17:54:15 2017-date Wed May 10 17:51:58 2017 s - exit: 21504

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

davidlange6 commented:
my question is a different one. Are the vector lengths known and fixed (for a given code version)? Answer: yes

ecalLUT.push_back(std::vector< std::vector< std::vector< uint32_t > > >(28, std::vector< std::vector< uint32_t > >(2, std::vector< uint32_t >(256))));

good.

so the vector<vector<vector<vector<uint32> should instead be

std::vector< std::array< std::array< std::array<uint32, 256>, 2 >, 28> > ecalLUT; (or maybe the number of phi bins is known too, but it seems not)

but btw - how was L1TCaloLayer1FetchLUTs not writing ecalLUT data into outer space previously? It seems to initialize the vector hierarchy only in your new version..

> On May 10, 2017, at 4:09 PM, rekovic <notifications@github.com> wrote:
> 
> @davidlange6
> Follow up on your question #18622 (comment).
> 
> Example of ecal and hcal the Layer1 Calib values are provided in eta (28 towers) x 2 sides x phi (256).
> Will document this also.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

rekovic commented:
please test with cms-sw/cmsdist#3017

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3017

cmsbuild commented:
+1
Tested at: 1418195478ee6e9ece0864d9b835c5e31c317c55
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19765/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19765/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 1672 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1830172
 - DQMHistoTests: Total failures: 54030
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1775962
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

rekovic committed:
Optimisation: replace multilayers of vectors by layers of arrays of known sizes.

cmsbuild commented:
Pull request #18622 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @rekovic, @franzoni, @ggovi, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

rekovic commented:
@davidlange6   Addressed your comments above.
As requested, changed multi-layers of vectors to layers of std::arrays with known sizes.

>but btw - how was L1TCaloLayer1FetchLUTs not writing ecalLUT data into outer space previously? It seems to initialize the vector hierarchy only in your new version..

Actually, the previous version did have the memory allocated. The vectors were initialised in the constructors.

rekovic commented:
please test with cms-sw/cmsdist#3017

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3017

rekovic commented:
+1

rekovic committed:
Clean up - remove commented lines.

rekovic commented:
please test with cms-sw/cmsdist#3017

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3017

rekovic commented:
+1

cmsbuild commented:
Pull request #18622 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @rekovic, @franzoni, @ggovi, @mmusich, @mulhearn, @davidlange6 can you please check and sign again.

cmsbuild commented:
+1
Tested at: 458bba8aed164e42747585e9cd62a6e90caf916e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19834/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18622/19834/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3363 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1829394
 - DQMHistoTests: Total failures: 34688
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1794526
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

ggovi commented on CondFormats/L1TObjects/interface/L1GtDefinitions.h:
```diff
@@ -136,7 +138,8 @@ enum L1GtConditionCategory {
     CondHfRingEtSums,
     CondBptx,
-    CondExternal
+    CondExternal,
```
@rekovic 
Are these extended enums used in the persistent data?  I mean, if the possible enum's values are stored in the DB, the previous releases won't be able to read the new data.

ghellwig replied:
@rekovic @ggovi from AlCa side we are fine with this PR since this file is the only one in our domain.
We'll sign, once we have received a satisfactory answer to this question.

rekovic commented:
@ggovi, @ghellwig  Thanks.
> Are these extended enums used in the persistent data? I mean, if the possible enum's values are stored in the DB, the previous releases won't be able to read the new data.

No.  These enums are not stored in the DB.

ghellwig commented:
+1
thanks @rekovic

rekovic commented:
@ggovi, @ghellwig, can you also please sign the 91x version #18671 (same code, same branch) ?

ggovi commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

rekovic commented:
@smuzaffar @davidlange6 
Need cms-sw/cmsdist#3017 merged in 92x.  It already is in 91x.

davidlange6 commented:
+1

davidlange6 commented:
hi @rekovic - looks like there is one workflow failing due to this - please provide a fix (its workflow 281)

Exception Message:
python encountered the error: <type 'exceptions.RuntimeError'>
edm::FileInPath unable to find file L1Trigger/L1TCalorimeter/data/lut_towEtThresh_dummy.txt anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /build/cmsbld/jenkins-workarea/workspace/ib-run-relvals/CMSSW_9_2_X_2017-05-16-2300/src:/build/cmsbld/jenkins-workarea/workspace/ib-run-relvals/CMSSW_9_2_X_2017-05-16-2300/external/slc6_amd64_gcc530/data:/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-05-16-2300/src:/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-05-16-2300/external/slc6_amd64_gcc530/data
Current directory is: /build/cmsbld/jenkins-workarea/workspace/ib-run-relvals/CMSSW_9_2_X_2017-05-16-2300/pyRelval/281.0/281.0_EPOS_PPb_8160GeV_MinimumBias+EPOS_PPb_8160GeV_MinimumBias+DIGIUP15_PPb+RECOUP15_PPb+HARVESTUP15_PPb

----- End Fatal Exception -------------------------------------------------

rekovic commented:
Hi @davidlange6 

It is a LUT file that is still not used in any p-p workflows.
However the workflow with -era Run2_2016_pA picks it
up. Sorry I missed it in the last externals PR.
This is fixed with cms-sw/cmsdist#3054
